frontend/packages/dev-console/src/components/health-checks/AddHealthCheckPage.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { RouteComponentProps } from 'react-router-dom';
+import { PageHeading, FirehoseResource, Firehose } from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { ServiceModel } from '@console/knative-plugin';
+import AddHealthCheckWrapper from './AddHealthCheckWrapper';
+
+type AddHealthCheckProps = RouteComponentProps<{
+  ns: string;
+  kind: string;
+  name: string;
+  containerName: string;
+}>;
+
+const AddHealthCheckPage: React.FC<AddHealthCheckProps> = ({ match }) => {
+  const { ns, kind, name, containerName } = match.params;
+  const resource: FirehoseResource[] = [
+    ...(kind === ServiceModel.kind
```

Comment:
not sure it'll be good if we can have resolved kind in params then we can avoid condition below

Reply:
We need this condition here. otherwise, firehose will not fetch D and DC.